File: Training-Site/Programmers/CodingTest/두-개-뽑아서-더하기/source.py
# https://programmers.co.kr/learn/courses/30/lessons/68644


def solution(numbers):
    answer = []

    # 인덱스 하나씩 구해옴
    for i in range(len(numbers)):
        for ii in range(len(numbers)):
            result = numbers[i]+numbers[ii]
            # 같은 인덱스의 값끼리는 더하지 않고, 같은 값이 포함되지 않을 때 answer에 추가
            if i != ii and result not in answer:
                answer.append(result)

    # 정렬
    answer.sort()
    return answer

# Test Case
print("Answer :", solution([5,0,2,7])) # Answer : [2,5,7,9,12]
print("Answer :", solution([2,1,3,4,1])) # Answer : [2,3,4,5,6,7]

File: Training-Site/Programmers/CodingTest/두-개-뽑아서-더하기/source.js
// https://programmers.co.kr/learn/courses/30/lessons/68644

function solution(numbers) {
  var answer = [];

  for (let i = 0; i <= numbers.length; i++) {
    for (let ii = 0; ii <= numbers.length; ii++) {
      var result = numbers[i] + numbers[ii];
      if (i != ii && !isNaN(result)) {
        answer.push(result);
      }
    }
  }

  // 중복 숫자 제거
  let uniqueArray = [];
  answer.forEach((element) => {
    if (!uniqueArray.includes(element)) {
      uniqueArray.push(element);
    }
  });

  // 오름차순 정렬
  uniqueArray.sort(function (a, b) {
    return a - b;
  });

  console.log(uniqueArray);
  return uniqueArray;
}

solution([2, 1, 3, 4, 1]);
